### ISSUE TITLE
Fall back to the default webhook profile image when the "override_icon_url" image fails to load

### ISSUE THREAD

**Issue by esethna** [mattermost/mattermost#10583]
If a webhook specifies a custom override_icon_url, but the image fails to load, we end up with an ugly visual experience:

![image](https://user-images.githubusercontent.com/13119842/55825753-9b810000-5ad4-11e9-98a6-d3ffaebd4fa2.png)

Instead of failing to load the image, the client should fallback to the default image:

![image](https://user-images.githubusercontent.com/13119842/55825775-a5a2fe80-5ad4-11e9-841d-dd848ca2ebc1.png)

Note that overriding icons requires setting `ServiceSettings.EnablePostIconOverride` to be `true`.

Fixing this issue will requires changes in both the mobile and webapp.

-- 

If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-15023

**Comment by bdsweet** [comment]
I'll work on this.

**Comment by hanzei** [comment]
Thanks @bdsweet. Let me know, if you have questions.

**Comment by bdsweet** [comment]
@hanzei - the image load issue happens both for a successful post and for an ephemeral post in the case the webhook fails and sends a "system_ephemeral" broadcast. 

For the ephemeral post, the web client attempts to use the override_icon_url in the ephemeral_message, but on the android mobile client it uses the system icon instead. What's the desired behaviour for the failure case?

**Comment by hanzei** [comment]
> but on the android mobile client it uses the system icon instead. What's the desired behaviour for the failure case?

The expected behaviour is the usage of `override_icon_url'. Could you please double check that the mobile client us behaving different and provide steps to reproduce? This seams like a bug.

**Comment by bdsweet** [comment]
Yeah, it's easy to reproduce. Just set up a slash-command like a GET to "example.com" which will return HTML that's not in the expected format. Then run the slash-command from the mobile app. It will show the system icon instead of the override icon since in notification.js it checks the system flag first, whereas on the web in post_profile_picture.jsx it only uses the system icon if the message has the system flag but is not a webhook.

I'll just fix that logic along with this issue.

**Comment by bdsweet** [comment]
Sorry for the delay - back on this now. Just need to get the end-to-end tests to run correctly and I'll submit a pull request

**Comment by hanzei** [comment]
Thanks for the heads up. Looking forward to your PR @bdsweet :+1:

**Comment by crspeller** [comment]
@bdsweet Need any help?

**Comment by hanzei** [comment]
Making this one available for the public again due to inactivity.

**Comment by panoramix360** [comment]
I was testing this issue to try to reproduce it and maybe fix it, but it seems it's already fixed? just tried here.

Take a look at the Android app:

<img width="254" alt="Image" src="https://github.com/user-attachments/assets/69a23a75-e410-433b-afba-0b749ab5b09c" />

**Comment by lieut-data** [comment]
Thanks for helping to reproduce! Closing as such, feel free to open a new ticket if needed.
